exporter/loadbalancingexporter/trace_exporter.go

```diff
@@ -88,80 +149,134 @@ func (e *traceExporterImp) Shutdown(context.Context) error {
 }
 
 func (e *traceExporterImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	var err error
 	var errs error
-	batches := batchpersignal.SplitTraces(td)
-	for _, batch := range batches {
-		errs = multierr.Append(errs, e.consumeTrace(ctx, batch))
+	var batches = make(map[string][]ptrace.Traces)
+	if e.routingKey == resourceAttrRouting {
+		batches, err = SplitTracesByResourceAttr(td, e.resourceAttrKeys)
+		if err != nil {
+			return err
+		}
+	} else {
+		batches["traceId"] = batchpersignal.SplitTraces(td)
+	}
+	rfs := make(map[string]routingFunction)
+	for key := range batches {
+		if key == "traceId" {
+			rfs[key] = func(x ptrace.Traces) (map[string][]int, error) {
+				return routeByTraceId(x)
+			}
+		} else {
+			rfs[key] = func(x ptrace.Traces) (map[string][]int, error) {
+				return routeByResourceAttr(x, key)
+			}
+		}
+	}
+	for key, tb := range batches {
+		for _, t := range tb {
+			errs = multierr.Append(errs, e.consumeTrace(ctx, t, rfs[key]))
+		}
 	}
 
 	return errs
 }
 
-func (e *traceExporterImp) consumeTrace(ctx context.Context, td ptrace.Traces) error {
+func (e *traceExporterImp) consumeTrace(ctx context.Context, td ptrace.Traces, rf routingFunction) error {
 	var exp component.Component
-	routingIds, err := routingIdentifiersFromTraces(td, e.routingKey)
+	routingIds, err := rf(td)
+	if err != nil {
+		return err
+	}
+	var rid string
+	for key := range routingIds {
+		rid = key
+		break
```

Comment:
Can you add a comment about this part? Just by looking at it, I can't tell what it's actually doing. If you care about only the first key, why do you need a loop?